web/react/components/create_comment.jsx

```diff
@@ -40,163 +44,194 @@ module.exports = React.createClass({
-                }
-                else {
+                if (err.message === 'Invalid RootId parameter') {
+                    if ($('#post_deleted').length > 0) {
+                        $('#post_deleted').modal('show');
+                    }
+                } else {
                     this.setState(state);
                 }
             }.bind(this)
         );
     },
     commentMsgKeyPress: function(e) {
-        if (e.which == 13 && !e.shiftKey && !e.altKey) {
+        if (e.which === 13 && !e.shiftKey && !e.altKey) {
             e.preventDefault();
             this.refs.textbox.getDOMNode().blur();
             this.handleSubmit(e);
         }
 
-        var t = new Date().getTime();
+        var t = Date.now();
         if ((t - this.lastTime) > 5000) {
-            SocketStore.sendMessage({channel_id: this.props.channelId, action: "typing", props: {"parent_id": this.props.rootId} });
+            SocketStore.sendMessage({channel_id: this.props.channelId, action: 'typing', props: {'parent_id': this.props.rootId}});
             this.lastTime = t;
         }
     },
     handleUserInput: function(messageText) {
         var draft = PostStore.getCommentDraft(this.props.rootId);
         if (!draft) {
-            draft = { previews: [], uploadsInProgress: 0};
+            draft = {previews: [], uploadsInProgress: []};
         }
         draft.message = messageText;
         PostStore.storeCommentDraft(this.props.rootId, draft);
 
-        $(".post-right__scroll").scrollTop($(".post-right__scroll")[0].scrollHeight);
-        $(".post-right__scroll").perfectScrollbar('update');
+        $('.post-right__scroll').scrollTop($('.post-right__scroll')[0].scrollHeight);
+        $('.post-right__scroll').perfectScrollbar('update');
         this.setState({messageText: messageText});
     },
-    handleFileUpload: function(newPreviews) {
+    handleUploadStart: function(clientIds, channelId) {
         var draft = PostStore.getCommentDraft(this.props.rootId);
         if (!draft) {
-            draft = { message: '', uploadsInProgress: 0, previews: []}
+            draft = {};
+            draft['message'] = '';
+            draft['uploadsInProgress'] = [];
+            draft['previews'] = [];
         }
 
-        $(".post-right__scroll").scrollTop($(".post-right__scroll")[0].scrollHeight);
-        $(".post-right__scroll").perfectScrollbar('update');
-        var previews = this.state.previews.concat(newPreviews);
-        var num = this.state.uploadsInProgress;
+        draft['uploadsInProgress'] = draft['uploadsInProgress'].concat(clientIds);
+        PostStore.storeCommentDraft(this.props.rootId, draft);
 
-        draft.previews = previews;
-        draft.uploadsInProgress = num-1;
+        this.setState({uploadsInProgress: draft['uploadsInProgress']});
+    },
+    handleFileUploadComplete: function(filenames, clientIds, channelId) {
+        var draft = PostStore.getCommentDraft(this.props.rootId);
+        if (!draft) {
```

Comment:
I was definitely thinking of that when I was writing this. I need to make sure that there's nowhere that depends on the behaviour of returning null, but I'll add that in.